util-flywayinstaller/src/main/java/com/github/cafapi/util/flywayinstaller/Migrator.java

```diff
@@ -0,0 +1,117 @@
+            + " allowDBDeletion: {}\n"
+            + " connectionString: {}\n"
+            + " dbName: {}\n"
+            + " username: {}\n"
+            + " password: {}", allowDBDeletion, connectionString, dbName, username, password);
+
+        LOGGER.info("Starting migration ...");
+        try (final BasicDataSource dbSource = new BasicDataSource()) {
+            dbSource.setUrl(checkAndConvertConnectionUrl(connectionString));
+            dbSource.setUsername(username);
+            dbSource.setPassword(password);
+
+            final boolean exists = checkDBExists(dbSource, dbName);
+            LOGGER.debug("Database exists: {}", exists);
+            if (!exists || allowDBDeletion) {
+                resetOrCreateDatabase(dbSource, exists, dbName);
+            }
+
+            LOGGER.info("About to perform DB update.");
+            final Flyway flyway = Flyway.configure()
+                .dataSource(dbSource.getUrl() + dbName, username, password)
+                .baselineOnMigrate(true)
+                .load();
+            flyway.migrate();
+            flyway.validate();
+            LOGGER.info("DB update finished.");
+        }
+        LOGGER.info("Migration completed ...");
+    }
+
+    private static void resetOrCreateDatabase(
+        final BasicDataSource dbSource,
+        final boolean exists,
+        final String dbName) throws SQLException
+    {
+        final String savedUrl = dbSource.getUrl();
+        dbSource.setUrl(dbSource.getUrl() + "postgres");
+        LOGGER.debug(dbSource.getUrl());
+        try (final Connection connection = dbSource.getConnection();
+             final Statement statement = connection.createStatement()) {
+            if (exists) {
+                LOGGER.info("force deletion has been specified.\nDeleting database {}", dbName);
+                statement.executeUpdate("DROP DATABASE " + dbName);
+                LOGGER.info("DELETED database: {}", dbName);
+            }
+            statement.executeUpdate("CREATE DATABASE " + dbName);
+            LOGGER.info("Created new database: {}", dbName);
+        }
+        dbSource.setUrl(savedUrl);
+    }
+
+    private static boolean checkDBExists(final BasicDataSource dbSource, final String dbName) throws SQLException
+    {
+        try (final Connection connection = dbSource.getConnection();
+             final Statement statement = connection.createStatement()) {
+            final boolean exists = statement.executeQuery("SELECT * FROM pg_database WHERE datname=lower('" + dbName + "');").next();
+            LOGGER.info("Database {} exists: {}", dbName, exists);
+            return exists;
+        }
+    }
+
+    private static String checkAndConvertConnectionUrl(final String connectionUrl) throws InvalidConnectionStringException
+    {
+        final String connectionUrlWithSlashes = connectionUrl.replace("\\", "/");
+        final Matcher matcher = Pattern.compile(CONNECTION_URL_REGEX).matcher(connectionUrlWithSlashes);
```

Comment:
What is this all about?  What are we trying to achieve here?

Reply:
Sometimes, we could receive an url like: `jdbc:postgresql://localhost:5437/?adaptiveFetch=false&adaptiveFetchMaximum=-1&adaptiveFetchMinimum=0`
This is to unsure that we get the right format and then to split it to `jdbc:postgresql://localhost:5437/`